### PR TITLE
Generalize website generation

### PR DESCRIPTION
**Pull request by frankharkins**
### Summary

Adds the ability to generate HTML that can be embedded into a markdown document.
You may find it easier to review each commit separately.

- (3ec7a4818f56de487638a245caeb6bd639a430e8) Makes the license field optional. A license is still a requirement for ecosystem projects, this change just allows building the website if it isn't set in the database.
- (d24a4ab3720d889db3131a00e22e301a8dc05756) Add 'standalone' option to generate a page that can be embedded in another page.
  This basically means exluding the head/body tags, removing some CSS, and removing indentation (to play nicely with markdown).
- (92000f26903a3d5bcf05c3f4d08d12e52c67eb1f) Specify output filename (not just folder) in case you don't want the page to be called `index.html`.
- (e98f6e5e5a26744e29157142b5ab58d8f7bf9dbf) Add custom css capabilities. This lets users add custom CSS rules, specified in the `website.toml` file.

### CODE REVIEW

File: ecosystem/html_templates/style.css.jinja [hunk 1]
```diff
@@ -92,10 +94,12 @@ cds-tag {
   margin-bottom: 8px;
 }
 
+{% if standalone %}
 p {
   font-size: 14px;
   color: #525252;
 }
+{% endif %}
```
Comment (Eric-Arellano) [inline ecosystem/html_templates/style.css.jinja:102]: I think it'd be helpful to group all the standalone CSS rules together.

File: ecosystem/html_templates/webpage.html.jinja [hunk 2]
```diff
@@ -8,6 +9,19 @@
     />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <title>{{ header.title.bold }} {{ header.title.normal }}</title>
+    <link
```
Comment (Eric-Arellano) [inline ecosystem/html_templates/webpage.html.jinja:12]: Maybe comment why these two CSS files are only for standalone.

File: ecosystem/html_templates/style.css.jinja (hunk 1 above)
Comment (frankharkins) [reply ecosystem/html_templates/style.css.jinja]: Good call, I can't move the media rules to the top as it breaks the reactivity, but I've moved the `p` rule.

File: ecosystem/html_templates/webpage.html.jinja (hunk 2 above)
Comment (frankharkins) [reply ecosystem/html_templates/webpage.html.jinja:14]: Done in 35eb19256065b4b09bfe4b76acde0b3663ab093b, thanks.

File: ecosystem/cli/website.py [hunk 3]
```diff
@@ -49,21 +57,28 @@ def _load_from_file(
...
     }
-    return projects, web_data, label_descriptions, templates
+    return projects, web_data, label_descriptions, templates, custom_css
 
 
 def _build_html(projects, web_data, label_descriptions, templates) -> str:
     """
     Take all data needed to build the website and produce a HTML string.
     """
+    # pylint: disable=too-many-locals
```
Comment (frankharkins) [inline ecosystem/cli/website.py:81]: Pylint is right to complain here but I think it should be fixed by refactoring other parts of the function which is out of scope of this PR

**Pull request merged**